process/block/sovereignChainBlock.go

```diff
@@ -0,0 +1,1279 @@
+	hdrsForCurrentBlock := make([]data.HeaderHandler, 0)
+
+	scbp.hdrsForCurrBlock.mutHdrsForBlock.RLock()
+	for _, headerInfo := range scbp.hdrsForCurrBlock.hdrHashAndInfo {
+		hdrsForCurrentBlock = append(hdrsForCurrentBlock, headerInfo.hdr)
+	}
+	scbp.hdrsForCurrBlock.mutHdrsForBlock.RUnlock()
+
+	process.SortHeadersByNonce(hdrsForCurrentBlock)
+
+	return hdrsForCurrentBlock
+}
+
+func (scbp *sovereignChainBlockProcessor) verifyCrossShardMiniBlockDstMe(sovereignChainHeader data.SovereignChainHeaderHandler) error {
+	miniBlockExtendedShardHeaderHashes, err := scbp.getAllMiniBlockDstMeFromExtendedShardHeaders(sovereignChainHeader)
+	if err != nil {
+		return err
+	}
+
+	crossMiniBlockHashes := sovereignChainHeader.GetMiniBlockHeadersWithDst(core.SovereignChainShardId)
+	for hash := range crossMiniBlockHashes {
+		if _, ok := miniBlockExtendedShardHeaderHashes[hash]; !ok {
+			return process.ErrCrossShardMBWithoutConfirmationFromNotifier
+		}
+	}
+
+	return nil
+}
+
+func (scbp *sovereignChainBlockProcessor) getAllMiniBlockDstMeFromExtendedShardHeaders(sovereignChainHeader data.SovereignChainHeaderHandler) (map[string][]byte, error) {
+	lastCrossNotarizedHeader, _, err := scbp.blockTracker.GetLastCrossNotarizedHeader(core.SovereignChainShardId)
+	if err != nil {
+		return nil, err
+	}
+
+	miniBlockExtendedShardHeaderHashes := make(map[string][]byte)
+
+	scbp.hdrsForCurrBlock.mutHdrsForBlock.RLock()
+	for _, extendedShardHeaderHash := range sovereignChainHeader.GetExtendedShardHeaderHashes() {
+		headerInfo, ok := scbp.hdrsForCurrBlock.hdrHashAndInfo[string(extendedShardHeaderHash)]
+		if !ok {
+			continue
+		}
+		shardHeaderExtended, ok := headerInfo.hdr.(*block.ShardHeaderExtended)
+		if !ok {
+			continue
+		}
+		if shardHeaderExtended.GetRound() <= lastCrossNotarizedHeader.GetRound() {
+			continue
+		}
+		if shardHeaderExtended.GetNonce() <= lastCrossNotarizedHeader.GetNonce() {
+			continue
+		}
+
+		incomingMiniBlocks := shardHeaderExtended.GetIncomingMiniBlocks()
+		for _, mb := range incomingMiniBlocks {
+			mbHash, errCalculateHash := core.CalculateHash(scbp.marshalizer, scbp.hasher, mb)
+			if errCalculateHash != nil {
+				return nil, errCalculateHash
+			}
+
+			miniBlockExtendedShardHeaderHashes[string(mbHash)] = extendedShardHeaderHash
+		}
+	}
+	scbp.hdrsForCurrBlock.mutHdrsForBlock.RUnlock()
+
+	return miniBlockExtendedShardHeaderHashes, nil
+}
+
+// applyBodyToHeader creates a miniblock header list given a block body
+func (scbp *sovereignChainBlockProcessor) applyBodyToHeader(
+	headerHandler data.HeaderHandler,
+	body *block.Body,
+) (*block.Body, error) {
+	sw := core.NewStopWatch()
+	sw.Start("applyBodyToHeader")
+	defer func() {
+		sw.Stop("applyBodyToHeader")
+		log.Debug("measurements", sw.GetMeasurements()...)
+	}()
+
+	var err error
+	err = headerHandler.SetMiniBlockHeaderHandlers(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	rootHash, err := scbp.accountsDB[state.UserAccountsState].RootHash()
```

Comment:
can reuse the baseProcess.getRootHash() function here

Reply:
Not the same. The one from baseProcess does a silent error getter. In this case, we return error